packages/app/src/app/pages/Pro/legacy-pages/WorkspacePlanSelection.tsx

```diff
@@ -1,591 +1,284 @@
     ...sortBy(
-      dashboard.teams.filter(t => t.id !== personalWorkspaceId),
-      [getUserAuthorization, 'name']
+      dashboard.teams.filter(team => team.id !== personalWorkspaceId),
+      team => team.name.toLowerCase()
     ),
-  ].filter(exists => exists);
+  ];
 
-  const activeWorkspace = workspaces.find(
-    workspace => workspace.id === activeTeam
-  );
-  const activeUserAuthorization =
-    activeWorkspace && getUserAuthorization(activeWorkspace);
-
-  React.useEffect(
-    function switchToWorkspaceWithAdminRights() {
-      // if type is PERSONAL_PRO, switch to personal account
-      if (type === SubscriptionType.PersonalPro) {
-        setActiveTeam({ id: personalWorkspaceId });
-        return;
-      }
-
-      // if you land on an account where you are not the admin
-      // switch accounts to one where you are an admin
-      // if none, switch to personal account
-      if (
-        activeUserAuthorization !== TeamMemberAuthorization.Admin &&
-        dashboard.teams.length
-      ) {
-        const workspaceWithAdminRights = dashboard.teams.find(
-          team =>
-            team.id !== personalWorkspaceId &&
-            getUserAuthorization(team) === TeamMemberAuthorization.Admin
-        );
-
-        if (workspaceWithAdminRights) {
-          setActiveTeam({ id: workspaceWithAdminRights.id });
-        } else {
-          setActiveTeam({ id: personalWorkspaceId });
-        }
-      }
-    },
-    [
-      type,
-      activeUserAuthorization,
-      dashboard.teams,
-      setActiveTeam,
-      personalWorkspaceId,
-      getUserAuthorization,
-    ]
-  );
-
-  React.useEffect(
-    function trackingPlanChoice() {
-      track('Pro - Set billing interval', { billingInterval });
+  const personalProCta: React.ComponentProps<typeof SubscriptionCard>['cta'] = {
+    text: 'Manage subscription',
+    onClick: () => {
+      modalOpened({ modal: 'legacyPayment' });
     },
-    [billingInterval]
-  );
-
-  React.useEffect(
-    function trackingWorkspace() {
-      track('Pro - Change Team selection');
-    },
-    [activeWorkspace]
-  );
-
-  if (!activeTeam || !dashboard.teams.length || !selectedPlan) return null;
-
-  const numberOfEditors = activeWorkspace.userAuthorizations.filter(
-    ({ authorization }) => authorization !== TeamMemberAuthorization.Read
-  ).length;
-  updateSeats(numberOfEditors);
-
-  const isLegacyPersonalPro = isPersonalWorkspace && user.subscription;
-  const currentSubscription = activeTeamInfo?.subscription;
-
-  // if there is mismatch of intent - team/personal
-  // or you don't have access to upgrade
-  // open the account switcher on load
-  const switcherDefaultOpen =
-    (type === SubscriptionType.TeamPro && isPersonalWorkspace) ||
-    activeUserAuthorization !== TeamMemberAuthorization.Admin;
+    variant: 'light',
+  };
+
+  const teamProCta: React.ComponentProps<
+    typeof SubscriptionCard
+    // eslint-disable-next-line no-nested-ternary
+  >['cta'] = isTeamAdmin
```

Comment:
I wouldn't encourage using nested ternary, but taking into account that this page will also be removed soon, it's okay

Reply:
I agree, although I thought it was pretty readable and laziness got the better of me. Glad I can keep it 😛 for now.